packages/charts/react-charting/src/components/HeatMapChart/HeatMapChart.base.tsx

```diff
@@ -601,38 +601,48 @@ export class HeatMapChartBase extends React.Component<IHeatMapChartProps, IHeatM
-          }
-          if (this._xAxisType === XAxisTypes.StringAxis) {
-            datapoint.x = this._getFormattedLabelForXAxisDataPoint(datapoint.x as string);
-          }
-          if (this._yAxisType === YAxisType.DateAxis) {
-            datapoint.y = this._getStringFormattedDate(datapoint.y as string, yAxisDateFormatString);
-          }
-          if (this._yAxisType === YAxisType.NumericAxis) {
-            datapoint.y = this._getStringFormattedNumber(datapoint.y as string, yAxisNumberFormatString);
-          }
-          if (this._yAxisType === YAxisType.StringAxis) {
-            datapoint.y = this._getFormattedLabelForYAxisDataPoint(datapoint.y as string);
-          }
-        });
+      this.props.sortOrder !== 'None'
+        ? yPoints[item]
+            .sort((a: IHeatMapChartDataPoint, b: IHeatMapChartDataPoint) => {
+              if (this._xAxisType === XAxisTypes.StringAxis) {
+                return this.props.sortOrder === 'None'
+                  ? 0
+                  : (a.x as string).toLowerCase() > (b.x as string).toLowerCase()
+                  ? 1
+                  : -1;
+              } else if (this._xAxisType === XAxisTypes.DateAxis) {
+                return (a.x as Date).getTime() - (b.x as Date).getTime();
+              } else if (this._xAxisType === XAxisTypes.NumericAxis) {
+                return +(a.x as string) > +(b.x as string) ? 1 : -1;
+              } else {
+                return a.x > b.x ? 1 : -1;
+              }
+            })
+            .forEach((datapoint: IHeatMapChartDataPoint) => {
+              if (this._xAxisType === XAxisTypes.DateAxis) {
+                datapoint.x = this._getStringFormattedDate(datapoint.x as string, xAxisDateFormatString);
+              }
+              if (this._xAxisType === XAxisTypes.NumericAxis) {
+                datapoint.x = this._getStringFormattedNumber(datapoint.x as string, xAxisNumberFormatString);
+              }
+              if (this._xAxisType === XAxisTypes.StringAxis) {
+                datapoint.x = this._getFormattedLabelForXAxisDataPoint(datapoint.x as string);
+              }
+              if (this._yAxisType === YAxisType.DateAxis) {
+                datapoint.y = this._getStringFormattedDate(datapoint.y as string, yAxisDateFormatString);
+              }
+              if (this._yAxisType === YAxisType.NumericAxis) {
+                datapoint.y = this._getStringFormattedNumber(datapoint.y as string, yAxisNumberFormatString);
+              }
+              if (this._yAxisType === YAxisType.StringAxis) {
+                datapoint.y = this._getFormattedLabelForYAxisDataPoint(datapoint.y as string);
+              }
+            })
+        : yPoints[item].forEach((datapoint: IHeatMapChartDataPoint) => {
+            if (this._xAxisType === XAxisTypes.StringAxis) {
+              datapoint.x = this._getFormattedLabelForXAxisDataPoint(datapoint.x as string);
+            }
```

Comment:
this code looks redundant

Reply:
No, this is not redundant. this._getFormattedLabelForXAxisDataPoint , this should run even if the sorting is not needed for 'None' case.

Reply:
No, this is not redundant. this._getFormattedLabelForXAxisDataPoint, this should run even if sorting is not required in case of 'none' case.